.github/workflows/generate_ipynb_files.yml

```diff
@@ -28,7 +26,10 @@ jobs:
 
       - name: Generate ipynb
         run: |
-          jupytext --to ipynb main/notebooks/**/**.py
+          for f in notebooks/**/*.py; do cp "$f" "${f%.py}_build_binary.py"; done
+          sed -i 's/^/!/' build_binaries.sh
+          sed -i $'/fetch prebuilt binaries/{r build_binaries.sh\n d}' notebooks/**/*_build_binary.py
```

Comment:
I suggest adding a comment explaining what is happening here and why do we want the script logic to be included in the resulting ipynb.